src/preload.py

```diff
@@ -609,8 +609,62 @@ def write_resin_device_pinning(app_data, output):
         )
 
 
+# Remove once supervisor is part of the hostapp
+def isSupervisor(app, images):
+    if len(app['services']) != 1:
+        # Supervisor is single service
+        return False
+
+    # Fetch image name from state app
+    for service in app['services']:
+        image_name = app['services'][service]['image']
+
+    for image in images:
+        if re.match(image, image_name):
+            return True
+    return False
+
+
 def write_apps_json(data, output):
-    """Writes data dict to output as json"""
+    """Updates or creates preloaded state"""
+    if os.path.isfile(output):
+        with open(output, "r") as f:
+            current_state = json.load(f)
+
+        # Fetch container image details from image
+        images, supervisor_version, balena_os_version = (
+            get_images_and_supervisor_version()
+        )
+
+        for uuid in list(current_state['apps'].keys()):
+            # Check state version consistency
+            if len(uuid) != len(list(data['apps'].keys())[0]):
+                log.error("The preloaded app state and the current state in"
+                          "the image have a version mismatch. This is likely"
+                          "a bug, please contact Balena's support")
```

Comment:
Could this error happen if:
* The user preloads an image file using an old version of the CLI, which overwrites `apps.json` in the image. The user saves a copy of the preloaded image file.
* "6 months later", the user upgrades the CLI to the latest version / this PR's version.
* The user attempts to preload the same image file again (not starting with a fresh image file) with the new CLI version.

Instead of "please contact Balena's support", how about:
```suggestion
                log.error("The preloaded app state and the current state in"
                          "the image have a version mismatch. Please use a fresh,"
                          "not-previously-preloaded image file.")
```
If the error still happens with a fresh image, the user would likely reach out to balena support or forums anyway. The CLI will also print this standard message when things fail:

```
$ balena preload ...
...
Additional information may be available with the `--debug` flag.

For further help or support, visit:
https://www.balena.io/docs/reference/balena-cli/#support-faq-and-troubleshooting
```

Reply:
Makes sense, I'll change it.